src/librbd/ObjectMap.h

```diff
@@ -69,6 +69,10 @@ class ObjectMap {
                   const ZTracer::Trace &parent_trace, T *callback_object) {
     assert(start_object_no < end_object_no);
     if (snap_id == CEPH_NOSNAP) {
+      if (end_object_no > m_object_map.size()) {
```

Comment:
Should it also check for ```start_object_no >= m_object_map.size()```?

Reply:
Taking that `start_object_no < end_object_no` is always true, if `start_object_no >= m_object_map.size()` is true,  `end_object_no > m_object_map.size()` will be also true, so it is enough to check only the last?

Reply:
... I guess I'm more concerned about the middle ground and ensuring that partial updates are handled. Cap ```m_end_object_no = std::min(m_end_object_no, m_object_map.size()``` and then skip if ````m_start_object_no >= m_object_map.size()```.

Reply:
I see. Thanks. Updated.